## Exercice 2.1 — Chargement des données (read_csv)
Objectif : Charger correctement des fichiers CSV dans des DataFrames. 

Instructions :

- Importez Pandas sous l'alias pd.
- Chargez rent_prediction.csv et employee_satisfaction.csv dans deux DataFrames distincts.
- Affichez la forme (shape) de chacun.<br> **Difficulté** : 🟢 Débutant

In [ ]:
# Import de Pandas sous l'alias pd
import pandas as pd

# Charegment des deux dataframes
data_rent = pd.read_csv("../datasets/rent_prediction.csv")
data_employee = pd.read_csv("../datasets/employee_satisfaction.csv")

In [2]:
# Affichage de shape pour rent
data_rent.shape

(510, 12)

In [4]:
# Affichage de shape pour employee
data_employee.shape

(510, 17)

## Exercice 2.2 — Aperçu rapide (head / tail)
Objectif : Prendre en main la structure d'un DataFrame. 

Instructions :

- Affichez les 10 premières lignes de employee_satisfaction.csv.
- Affichez les 10 dernières lignes.
- Notez par écrit (en commentaire) trois observations sur ce que vous voyez.<br> **Difficulté** : 🟢 Débutant

In [6]:
# Affichage des 10 premieres lignes de employee_satisfaction.csv
data_employee.head(10)

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
0,1,58.0,Homme,Marketing,12.0,Licence,2804177.0,Oui,13.0,95.0,8.0,0.0,3.0,Bon,Insatisfait,16.0,Oui
1,2,48.0,Femme,Marketing,NaN,Master,6000000.0,Non,12.0,48.0,10.0,4.0,1.0,Moyen,NaN,8.0,Non
2,3,34.0,Femme,Administration,16.0,NaN,4147951.0,Non,6.0,6.0,1.0,0.0,5.0,Très mauvais,Neutre,13.0,Non
3,4,27.0,Femme,Production,3.0,Licence,896874.0,Non,14.0,109.0,2.0,0.0,5.0,Moyen,Très insatisfait,16.0,Non
4,5,NaN,Femme,Administration,11.0,Master,3062453.0,Non,18.0,NaN,11.0,2.0,3.0,NaN,Très satisfait,28.0,Oui
5,6,58.0,Femme,Production,18.0,Licence,2658571.0,Non,34.0,91.0,11.0,2.0,2.0,Très mauvais,Insatisfait,30.0,Non
6,7,38.0,Homme,Production,11.0,Secondaire,1697179.0,Non,NaN,53.0,8.0,1.0,3.0,NaN,Satisfait,3.0,Oui
7,8,42.0,Homme,Commercial,8.0,Secondaire,1635366.0,Non,34.0,104.0,4.0,1.0,5.0,Moyen,Satisfait,17.0,Oui
8,9,30.0,Homme,Commercial,6.0,Licence,1890079.0,Oui,18.0,86.0,1.0,1.0,3.0,Bon,Insatisfait,26.0,Non
9,10,30.0,Femme,Ressources Humaines,11.0,Licence,2414085.0,Non,19.0,44.0,5.0,0.0,1.0,Bon,Satisfait,19.0,Oui


In [7]:
# Affichage des 10 dernieres lignes de employee_satisfaction.csv
data_employee.tail(10)

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
500,501,43.0,Femme,Recherche et Développement,12.0,Licence,2912806.0,Non,15.0,92.0,12.0,5.0,1.0,Excellent,Satisfait,NaN,Oui
501,502,40.0,Homme,Logistique,18.0,Licence,3273162.0,Oui,39.0,0.0,16.0,4.0,4.0,Moyen,Neutre,24.0,Non
502,503,53.0,Femme,Administration,6.0,Master,2230929.0,Non,4.0,73.0,6.0,1.0,5.0,NaN,Insatisfait,13.0,Oui
503,504,49.0,Homme,Logistique,14.0,Master,3501730.0,Non,1.0,98.0,9.0,0.0,3.0,Bon,Très insatisfait,24.0,Non
504,505,31.0,Femme,NaN,12.0,Licence,3389712.0,Oui,30.0,82.0,5.0,0.0,5.0,Bon,Neutre,3.0,Oui
505,506,40.0,Homme,NaN,10.0,Master,2586686.0,Non,19.0,76.0,0.0,0.0,1.0,Bon,Neutre,12.0,Oui
506,507,23.0,Homme,Informatique,0.0,Licence,579742.0,Non,21.0,81.0,0.0,1.0,4.0,Mauvais,Satisfait,5.0,Oui
507,508,45.0,Femme,Administration,NaN,Secondaire,338686.0,Non,24.0,58.0,0.0,1.0,2.0,NaN,Satisfait,8.0,Non
508,509,28.0,Femme,Finance,8.0,Secondaire,1426015.0,Non,38.0,91.0,0.0,1.0,1.0,Moyen,Satisfait,24.0,Non
509,510,51.0,Homme,Ressources Humaines,16.0,NaN,3820482.0,Non,10.0,35.0,14.0,2.0,3.0,Excellent,Neutre,8.0,Oui


## Exercice 2.3 — Informations générales (info)
Objectif : Obtenir une vue d'ensemble technique du dataset. 

Instructions :

- Utilisez .info() sur les deux DataFrames.
- Identifiez le nombre de valeurs non nulles par colonne.
- Identifiez les colonnes numériques et les colonnes catégorielles.<br> **Difficulté** : 🟢 Débutant

In [8]:
# Affichage de .info() pour rent_prediction.csv
data_rent.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 510 entries, 0 to 509
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   IdentifiantMaison      510 non-null    int64  
 1   Chambres               485 non-null    float64
 2   Salon                  485 non-null    object 
 3   SalleDeBainInterieure  485 non-null    object 
 4   Parking                485 non-null    object 
 5   Meuble                 485 non-null    object 
 6   Jardin                 485 non-null    object 
 7   Superficie_m2          485 non-null    float64
 8   DistanceRoute_m        485 non-null    float64
 9   Quartier               485 non-null    object 
 10  AgeMaison              485 non-null    float64
 11  LoyerMensuel_BIF       510 non-null    int64  
dtypes: float64(4), int64(2), object(6)
memory usage: 47.9+ KB


In [10]:
# Affichage de .info() pour employee_satisfaction.csv
data_employee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 510 entries, 0 to 509
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   ID_Employe              510 non-null    int64  
 1   Age                     485 non-null    float64
 2   Sexe                    485 non-null    object 
 3   Departement             485 non-null    object 
 4   Annees_Experience       485 non-null    float64
 5   Niveau_Etudes           485 non-null    object 
 6   Salaire_Mensuel_BIF     485 non-null    float64
 7   Travail_A_Distance      485 non-null    object 
 8   Heures_Supplementaires  485 non-null    float64
 9   Heures_Formation        485 non-null    float64
 10  Anciennete_Entreprise   485 non-null    float64
 11  Nombre_Promotions       485 non-null    float64
 12  Evaluation_Performance  485 non-null    float64
 13  Equilibre_Vie_Travail   485 non-null    object 
 14  Satisfaction_Salaire    485 non-null    ob

## Exercice 2.4 — Statistiques descriptives (describe)
Objectif : Résumer statistiquement un dataset. Instructions :

- Utilisez .describe() pour les colonnes numériques.
- Utilisez .describe(include='object') (ou équivalent) pour les colonnes catégorielles.
- Identifiez la colonne numérique ayant le plus grand écart-type relatif.<br> **Difficulté** : 🟢 Débutant

In [11]:
# Statistique descriptive pour les colonnes numériques
data_rent.describe()

,IdentifiantMaison,Chambres,Superficie_m2,DistanceRoute_m,AgeMaison,LoyerMensuel_BIF
count,510.000000,485.000000,485.000000,485.000000,485.000000,5.100000e+02
mean,255.500000,3.521649,168.587629,154.637113,19.934021,1.244608e+06
std,147.368586,1.707436,79.464914,85.262406,11.544729,7.004331e+05
min,1.000000,1.000000,30.000000,6.000000,0.000000,1.361490e+05
25%,128.250000,2.000000,103.000000,76.000000,10.000000,6.999250e+05
50%,255.500000,4.000000,171.000000,154.000000,20.000000,1.082640e+06
75%,382.750000,5.000000,235.000000,226.000000,30.000000,1.680108e+06
max,510.000000,6.000000,320.000000,300.000000,40.000000,2.600000e+06


In [12]:
# Statistique descriptive pour les colonnes catégorielles
data_rent.describe(include='object')

,Salon,SalleDeBainInterieure,Parking,Meuble,Jardin,Quartier
count,485,485,485,485,485,485
unique,2,2,2,2,2,14
top,Oui,Oui,Oui,Non,Non,Nyakabiga
freq,450,394,282,392,305,49


## Exercice 2.5 — Types de données
Objectif : Comprendre comment Pandas type chaque colonne. Instructions :

- Affichez dtypes pour les deux DataFrames.
- Expliquez pourquoi Departement n'est pas une colonne numérique.
- Identifiez une colonne dont le type pourrait être optimisé (par exemple float64 → int64 après nettoyage).<br> **Difficulté** : 🟢 Débutant

In [14]:
# Affichage de dtypes pour rent_prediction.csv
data_rent.dtypes

IdentifiantMaison          int64
Chambres                 float64
Salon                     object
SalleDeBainInterieure     object
Parking                   object
Meuble                    object
Jardin                    object
Superficie_m2            float64
DistanceRoute_m          float64
Quartier                  object
AgeMaison                float64
LoyerMensuel_BIF           int64
dtype: object

In [15]:
# Affichage de dtypes pour employee_satisfaction.csv
data_employee.dtypes

ID_Employe                  int64
Age                       float64
Sexe                       object
Departement                object
Annees_Experience         float64
Niveau_Etudes              object
Salaire_Mensuel_BIF       float64
Travail_A_Distance         object
Heures_Supplementaires    float64
Heures_Formation          float64
Anciennete_Entreprise     float64
Nombre_Promotions         float64
Evaluation_Performance    float64
Equilibre_Vie_Travail      object
Satisfaction_Salaire       object
Nombre_Absences           float64
Satisfait                  object
dtype: object

## Exercice 2.6 — Valeurs manquantes
Objectif : Détecter et quantifier les valeurs manquantes. 

Instructions :

- Comptez le nombre de valeurs manquantes par colonne avec isnull().sum().
- Calculez le pourcentage de valeurs manquantes par colonne.
- Proposez, pour chaque type de colonne (numérique / catégorielle), une stratégie de traitement (suppression, imputation par la moyenne/médiane/mode, etc.) et justifiez votre choix.<br> **Difficulté** : 🟡 Intermédiaire

In [17]:
# Affichage des valeurs manquantes pour rent_prediction.csv
data_rent.isnull().sum()

IdentifiantMaison         0
Chambres                 25
Salon                    25
SalleDeBainInterieure    25
Parking                  25
Meuble                   25
Jardin                   25
Superficie_m2            25
DistanceRoute_m          25
Quartier                 25
AgeMaison                25
LoyerMensuel_BIF          0
dtype: int64

In [18]:
# Affichage des valeurs manquantes pour emmployee_satisfaction.csv
data_employee.isnull().sum()

ID_Employe                 0
Age                       25
Sexe                      25
Departement               25
Annees_Experience         25
Niveau_Etudes             25
Salaire_Mensuel_BIF       25
Travail_A_Distance        25
Heures_Supplementaires    25
Heures_Formation          25
Anciennete_Entreprise     25
Nombre_Promotions         25
Evaluation_Performance    25
Equilibre_Vie_Travail     25
Satisfaction_Salaire      25
Nombre_Absences           25
Satisfait                  0
dtype: int64

In [19]:
# Pourcentage des valeurs nulles pour rent_prediction.csv
data_rent.isnull().mean() * 100

IdentifiantMaison        0.000000
Chambres                 4.901961
Salon                    4.901961
SalleDeBainInterieure    4.901961
Parking                  4.901961
Meuble                   4.901961
Jardin                   4.901961
Superficie_m2            4.901961
DistanceRoute_m          4.901961
Quartier                 4.901961
AgeMaison                4.901961
LoyerMensuel_BIF         0.000000
dtype: float64

In [20]:
# Pourcentage des valeurs nulles pour employee_satisfaction.csv
data_employee.isnull().mean() * 100

ID_Employe                0.000000
Age                       4.901961
Sexe                      4.901961
Departement               4.901961
Annees_Experience         4.901961
Niveau_Etudes             4.901961
Salaire_Mensuel_BIF       4.901961
Travail_A_Distance        4.901961
Heures_Supplementaires    4.901961
Heures_Formation          4.901961
Anciennete_Entreprise     4.901961
Nombre_Promotions         4.901961
Evaluation_Performance    4.901961
Equilibre_Vie_Travail     4.901961
Satisfaction_Salaire      4.901961
Nombre_Absences           4.901961
Satisfait                 0.000000
dtype: float64

## Stratégie de traitement des valeurs manquantes

Le choix de la méthode de traitement dépend de la nature de la variable (numérique ou catégorielle) ainsi que de la distribution des données.

### 1. Colonnes numériques

Les variables numériques représentent des quantités mesurables. Plusieurs stratégies peuvent être utilisées selon les caractéristiques des données.

#### Imputation par la moyenne

La moyenne est adaptée lorsque les données suivent une distribution symétrique et qu'elles ne contiennent pas de valeurs aberrantes importantes.

**Avantages :**
- Facile à mettre en œuvre.
- Conserve la taille du jeu de données.

#### Imputation par la médiane

La médiane est préférable lorsque la variable contient des valeurs aberrantes (outliers).

**Pourquoi ?**
Contrairement à la moyenne, la médiane est peu influencée par les valeurs extrêmes. Elle fournit une estimation plus robuste.

### 2. Colonnes catégorielles

Les variables catégorielles représentent des catégories ou des modalités .

#### Imputation par la mode

La stratégie la plus courante consiste à remplacer les valeurs manquantes par la **mode**, c'est-à-dire la catégorie la plus fréquente.

**Pourquoi ?**
- Elle conserve la catégorie la plus représentative du jeu de données.
- Elle évite la création d'une nouvelle catégorie artificielle.
- Elle est simple et efficace lorsque le pourcentage de valeurs manquantes est faible.

---

## Cas de suppression des données

Lorsque le nombre de valeurs manquantes est très élevé (par exemple supérieur à 40 % ou 50 %), il peut être préférable de supprimer la colonne si elle apporte peu d'informations. De même, les lignes contenant un grand nombre de valeurs manquantes peuvent être supprimées si leur conservation risque de dégrader la qualité des analyses.


## Exercice 2.7 — Valeurs dupliquées
Objectif : Garantir l'unicité des observations. 

Instructions :

- Vérifiez la présence de lignes dupliquées avec .duplicated().
- Si des doublons existent, supprimez-les avec .drop_duplicates().
- Vérifiez que la suppression n'a pas affecté la colonne d'identifiant.<br> **Difficulté** : 🟢 Débutant

In [23]:
# Vérifiaction de la présence des doublons dans rent_prediction.csv
data_rent.duplicated().sum()

0

In [26]:
# Vérifiaction de la présence des doublons dans employee_satisfaction.csv
data_employee.duplicated().sum()

0

In [28]:
# Supression des doublons
data_rent.drop_duplicates()
data_employee.drop_duplicates()

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
0,1,58.0,Homme,Marketing,12.0,Licence,2804177.0,Oui,13.0,95.0,8.0,0.0,3.0,Bon,Insatisfait,16.0,Oui
1,2,48.0,Femme,Marketing,NaN,Master,6000000.0,Non,12.0,48.0,10.0,4.0,1.0,Moyen,NaN,8.0,Non
2,3,34.0,Femme,Administration,16.0,NaN,4147951.0,Non,6.0,6.0,1.0,0.0,5.0,Très mauvais,Neutre,13.0,Non
3,4,27.0,Femme,Production,3.0,Licence,896874.0,Non,14.0,109.0,2.0,0.0,5.0,Moyen,Très insatisfait,16.0,Non
4,5,NaN,Femme,Administration,11.0,Master,3062453.0,Non,18.0,NaN,11.0,2.0,3.0,NaN,Très satisfait,28.0,Oui
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
505,506,40.0,Homme,NaN,10.0,Master,2586686.0,Non,19.0,76.0,0.0,0.0,1.0,Bon,Neutre,12.0,Oui
506,507,23.0,Homme,Informatique,0.0,Licence,579742.0,Non,21.0,81.0,0.0,1.0,4.0,Mauvais,Satisfait,5.0,Oui
507,508,45.0,Femme,Administration,NaN,Secondaire,338686.0,Non,24.0,58.0,0.0,1.0,2.0,NaN,Satisfait,8.0,Non
508,509,28.0,Femme,Finance,8.0,Secondaire,1426015.0,Non,38.0,91.0,0.0,1.0,1.0,Moyen,Satisfait,24.0,Non


## Exercice 2.8 — Filtrage
Objectif : Sélectionner des sous-ensembles de données selon des conditions. 

Instructions :

- Filtrez les employés du département de votre choix ayant répondu Satisfait == "Non".
- Filtrez les employés ayant plus de 45 ans et plus de 5 ans d'ancienneté.
- Sur rent_prediction.csv, filtrez les logements dont le loyer dépasse un seuil que vous définirez.<br> **Difficulté** : 🟡 Intermédiaire

In [33]:
# Filtrage des employes du departement ayant repondu Satisfait=Non
data_employee[(data_employee['Departement'] == "Logistique") & (data_employee['Satisfait'] == "Non")]

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
62,63,33.0,Femme,Logistique,3.0,Master,1066341.0,Oui,3.0,6.0,0.0,0.0,4.0,Bon,Insatisfait,15.0,Non
69,70,40.0,Homme,Logistique,18.0,Licence,3273162.0,Oui,39.0,0.0,NaN,4.0,4.0,Moyen,Neutre,24.0,Non
81,82,28.0,Homme,Logistique,0.0,Licence,NaN,Non,3.0,0.0,0.0,1.0,1.0,Bon,Insatisfait,19.0,Non
86,87,30.0,Homme,Logistique,8.0,Licence,1779815.0,Oui,31.0,60.0,7.0,3.0,5.0,Très mauvais,Insatisfait,10.0,Non
89,90,54.0,Femme,Logistique,5.0,Master,1786805.0,Non,34.0,23.0,5.0,2.0,4.0,Mauvais,Très insatisfait,13.0,Non
130,131,49.0,Femme,Logistique,19.0,Licence,3120899.0,Non,6.0,70.0,3.0,2.0,NaN,Mauvais,Insatisfait,22.0,Non
148,149,30.0,Homme,Logistique,12.0,Doctorat,3799363.0,Non,26.0,29.0,3.0,1.0,2.0,Moyen,Neutre,18.0,Non
159,160,25.0,Homme,Logistique,6.0,NaN,1328470.0,Non,20.0,8.0,NaN,1.0,2.0,Très mauvais,NaN,19.0,Non
222,223,59.0,Femme,Logistique,26.0,Master,5211855.0,Oui,21.0,68.0,12.0,2.0,2.0,Excellent,NaN,29.0,Non
232,233,35.0,Femme,Logistique,9.0,Master,2472155.0,Non,29.0,NaN,7.0,0.0,2.0,Moyen,Insatisfait,NaN,Non


In [36]:
# Filtrage des employes ayant plus de 45 ans et 5 ans d'experience
data_employee[(data_employee['Annees_Experience'] > 5) & (data_employee['Age'] > 45)]

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
0,1,58.0,Homme,Marketing,12.0,Licence,2804177.0,Oui,13.0,95.0,8.0,0.0,3.0,Bon,Insatisfait,16.0,Oui
5,6,58.0,Femme,Production,18.0,Licence,2658571.0,Non,34.0,91.0,11.0,2.0,2.0,Très mauvais,Insatisfait,30.0,Non
11,12,55.0,Femme,Support Client,30.0,Licence,4040781.0,Non,34.0,NaN,11.0,1.0,2.0,Excellent,Neutre,23.0,Oui
18,19,49.0,Homme,Administration,24.0,Licence,4720236.0,Non,NaN,53.0,20.0,3.0,4.0,Bon,Insatisfait,26.0,Oui
26,27,46.0,Femme,Administration,20.0,Master,4947611.0,Non,3.0,29.0,10.0,4.0,2.0,Moyen,Insatisfait,25.0,Non
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
484,485,51.0,Femme,Support Client,16.0,Doctorat,NaN,Non,6.0,48.0,8.0,3.0,2.0,Moyen,Insatisfait,12.0,Oui
488,489,56.0,Femme,Finance,13.0,Master,4572226.0,Oui,24.0,117.0,12.0,4.0,5.0,Moyen,Neutre,6.0,Oui
502,503,53.0,Femme,Administration,6.0,Master,2230929.0,Non,4.0,73.0,6.0,1.0,5.0,NaN,Insatisfait,13.0,Oui
503,504,49.0,Homme,Logistique,14.0,Master,3501730.0,Non,1.0,98.0,9.0,0.0,3.0,Bon,Très insatisfait,24.0,Non


In [39]:
# Filtrage des logements dont le loyer depasse un seuil
def affichage_logements(seuil: float):
    """Fonction retournant les logements dont le loyer depasse un seuil defini.

    Args:   
        seuil (float): Seuil à definir pour comparaison 

    Returns:
        DataFrame: Retourne un objet de type DataFrame
    """
    return data_rent[data_rent["LoyerMensuel_BIF"] > seuil]

affichage_logements(2000000)

,IdentifiantMaison,Chambres,Salon,SalleDeBainInterieure,Parking,Meuble,Jardin,Superficie_m2,DistanceRoute_m,Quartier,AgeMaison,LoyerMensuel_BIF
0,1,4.0,Oui,Oui,NaN,Non,Oui,192.0,277.0,Rohero,17.0,2600000
1,2,5.0,Oui,Oui,Non,Oui,Non,234.0,47.0,Rohero,37.0,2600000
3,4,5.0,Oui,Oui,Oui,Oui,Oui,227.0,174.0,Gasekebuye,22.0,2600000
4,5,5.0,Oui,Oui,Oui,NaN,Non,262.0,216.0,Gihosha,21.0,2085458
9,10,5.0,Oui,Oui,Oui,Oui,Non,238.0,115.0,Kiriri,31.0,2600000
...,...,...,...,...,...,...,...,...,...,...,...,...
491,492,6.0,Oui,Oui,Non,Non,Oui,254.0,53.0,Gasekebuye,40.0,2600000
492,493,6.0,Oui,Oui,Oui,Oui,Non,290.0,129.0,Ngagara,24.0,2043228
503,504,5.0,NaN,Oui,Oui,Non,Oui,257.0,122.0,Gasekebuye,22.0,2451878
507,508,4.0,Oui,Oui,Oui,Non,Oui,192.0,277.0,Rohero,17.0,2600000


## Exercice 2.9 — Tri (sort_values)
Objectif : Ordonner un DataFrame selon une ou plusieurs colonnes. 

Instructions :

- Triez les employés par Salaire_Mensuel_BIF décroissant et affichez le top 10.
- Triez par deux critères combinés (ex. Departement puis Salaire_Mensuel_BIF).
- Identifiez les 5 loyers les plus bas de rent_prediction.csv.<br> **Difficulté** : 🟢 Débutant

In [41]:
# Trier par salaire décroissant et afficher les 10 premiers employés
data_employee.sort_values(
    by='Salaire_Mensuel_BIF',
    ascending=False
).head(10)

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
128,129,51.0,Homme,Marketing,32.0,Doctorat,6000000.0,Non,20.0,89.0,13.0,5.0,4.0,Moyen,Satisfait,14.0,Oui
305,306,56.0,Femme,Finance,25.0,Master,6000000.0,Non,1.0,42.0,NaN,0.0,2.0,Mauvais,Satisfait,5.0,Oui
287,288,46.0,Femme,Informatique,24.0,Licence,6000000.0,Oui,24.0,109.0,10.0,1.0,1.0,Bon,Insatisfait,8.0,Oui
291,292,52.0,Femme,Administration,29.0,Master,6000000.0,Non,32.0,88.0,12.0,5.0,2.0,Bon,NaN,5.0,Oui
68,69,59.0,Homme,NaN,25.0,Master,6000000.0,Oui,14.0,32.0,11.0,3.0,5.0,Bon,Neutre,16.0,Oui
442,443,52.0,Femme,Finance,30.0,Master,6000000.0,Oui,19.0,106.0,7.0,2.0,1.0,Bon,Insatisfait,22.0,Oui
300,301,49.0,Femme,Informatique,28.0,Licence,6000000.0,Non,15.0,77.0,20.0,3.0,5.0,Mauvais,Neutre,25.0,Oui
433,434,NaN,Homme,NaN,16.0,Doctorat,6000000.0,Non,7.0,24.0,10.0,2.0,1.0,Très mauvais,Insatisfait,1.0,Non
310,311,52.0,Homme,Finance,25.0,Master,6000000.0,Non,25.0,24.0,24.0,2.0,2.0,NaN,Très satisfait,7.0,Oui
264,265,54.0,Femme,NaN,28.0,Master,6000000.0,Oui,31.0,66.0,19.0,6.0,1.0,Moyen,Très satisfait,19.0,Oui


In [42]:
# Trier par département (A→Z) puis par salaire décroissant
data_employee.sort_values(
    by=['Departement', 'Salaire_Mensuel_BIF'],
    ascending=[True, False]
)

,ID_Employe,Age,Sexe,Departement,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Travail_A_Distance,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait
163,164,56.0,Femme,Administration,26.0,Master,6000000.0,Non,15.0,118.0,15.0,6.0,5.0,Bon,Neutre,20.0,Oui
291,292,52.0,Femme,Administration,29.0,Master,6000000.0,Non,32.0,88.0,12.0,5.0,2.0,Bon,NaN,5.0,Oui
330,331,59.0,Femme,Administration,21.0,Doctorat,6000000.0,Oui,27.0,95.0,7.0,2.0,1.0,Très mauvais,Très satisfait,9.0,Oui
380,381,56.0,Femme,Administration,30.0,Master,6000000.0,Non,40.0,44.0,NaN,1.0,3.0,Moyen,Neutre,25.0,Non
284,285,57.0,Homme,Administration,25.0,Master,5528666.0,Non,19.0,9.0,8.0,NaN,1.0,Moyen,Satisfait,23.0,Non
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
98,99,27.0,Homme,NaN,2.0,Secondaire,709113.0,Non,11.0,120.0,0.0,0.0,4.0,Bon,Très insatisfait,19.0,Oui
309,310,54.0,Homme,NaN,1.0,Master,642338.0,Oui,11.0,42.0,1.0,0.0,3.0,Bon,Neutre,23.0,Oui
116,117,24.0,Homme,NaN,NaN,Licence,573746.0,Oui,13.0,104.0,1.0,1.0,3.0,Moyen,Très insatisfait,1.0,Non
451,452,48.0,NaN,NaN,2.0,Secondaire,545821.0,Non,21.0,96.0,0.0,1.0,4.0,Très mauvais,Neutre,10.0,Non


In [44]:
# Les 5 loyers les plus bas
data_rent.sort_values(
    by="LoyerMensuel_BIF",
    ascending=True
).head(5)

,IdentifiantMaison,Chambres,Salon,SalleDeBainInterieure,Parking,Meuble,Jardin,Superficie_m2,DistanceRoute_m,Quartier,AgeMaison,LoyerMensuel_BIF
139,140,1.0,Non,Non,Non,NaN,Non,49.0,205.0,Buyenzi,12.0,136149
234,235,1.0,Non,Non,Non,Non,Non,30.0,64.0,Bwiza,13.0,142096
45,46,6.0,Oui,Oui,Oui,Oui,Non,320.0,192.0,Kiriri,15.0,150000
273,274,1.0,Oui,Non,Non,Non,Non,32.0,NaN,Buyenzi,20.0,151335
187,188,1.0,Non,Oui,Non,Non,Non,32.0,148.0,Musaga,40.0,156174


## Exercice 2.10 — Groupby simple
Objectif : Agréger des données par catégorie. 

Instructions :

- Regroupez les employés par Departement et calculez le salaire moyen par groupe.
- Regroupez les logements par quartier et calculez le loyer moyen par groupe.
- Identifiez visuellement (sans graphique pour l'instant) le groupe le plus élevé et le plus bas dans chaque cas.<br> **Difficulté** : 🟡 Intermédiaire


In [45]:
# Salaire moyen par département
data_employee.groupby("Departement")["Salaire_Mensuel_BIF"].mean()

Departement
Administration                2.447688e+06
Commercial                    2.208388e+06
Finance                       3.260001e+06
Informatique                  2.823860e+06
Logistique                    2.516394e+06
Marketing                     2.503896e+06
Production                    2.012357e+06
Recherche et Développement    2.695685e+06
Ressources Humaines           3.040943e+06
Support Client                2.053510e+06
Name: Salaire_Mensuel_BIF, dtype: float64

In [47]:
# Loyer moyen par quartier
data_rent.groupby("Quartier")["LoyerMensuel_BIF"].mean().sort_values(ascending=False)

Quartier
Rohero        2.140383e+06
Kiriri        1.950949e+06
Gasekebuye    1.865619e+06
Gihosha       1.482161e+06
Kinanira      1.410618e+06
Ngagara       1.368809e+06
Nyakabiga     1.095939e+06
Cibitoke      9.961006e+05
Jabe          9.563693e+05
Musaga        8.901474e+05
Kamenge       7.669090e+05
Kinama        7.053754e+05
Buyenzi       6.066935e+05
Bwiza         5.986666e+05
Name: LoyerMensuel_BIF, dtype: float64

In [50]:
# Département avec le salaire moyen le plus élevé
highest_salary = data_employee.groupby("Departement")["Salaire_Mensuel_BIF"].mean().idxmax()
highest_salary_value = data_employee.groupby("Departement")["Salaire_Mensuel_BIF"].mean().max()

# Département avec le salaire moyen le plus faible
lowest_salary = data_employee.groupby("Departement")["Salaire_Mensuel_BIF"].mean().idxmin()
lowest_salary_value = data_employee.groupby("Departement")["Salaire_Mensuel_BIF"].mean().min()

print(f"Département avec le salaire moyen le plus élevé : {highest_salary} ({highest_salary_value:.2f} BIF)")
print(f"Département avec le salaire moyen le plus faible : {lowest_salary} ({lowest_salary_value:.2f} BIF)")

Département avec le salaire moyen le plus élevé : Finance (3260001.05 BIF)
Département avec le salaire moyen le plus faible : Production (2012357.11 BIF)


In [51]:
# Quartier avec le loyer moyen le plus élevé
highest_rent = data_rent.groupby("Quartier")["LoyerMensuel_BIF"].mean().sort_values(ascending=False).idxmax()
highest_rent_value = data_rent.groupby("Quartier")["LoyerMensuel_BIF"].mean().sort_values(ascending=False).max()

# Quartier avec le loyer moyen le plus faible
lowest_rent = data_rent.groupby("Quartier")["LoyerMensuel_BIF"].mean().sort_values(ascending=False).idxmin()
lowest_rent_value = data_rent.groupby("Quartier")["LoyerMensuel_BIF"].mean().sort_values(ascending=False).min()

print(f"Quartier avec le loyer moyen le plus élevé : {highest_rent} ({highest_rent_value:.2f} BIF)")
print(f"Quartier avec le loyer moyen le plus faible : {lowest_rent} ({lowest_rent_value:.2f} BIF)")

Quartier avec le loyer moyen le plus élevé : Rohero (2140382.79 BIF)
Quartier avec le loyer moyen le plus faible : Bwiza (598666.60 BIF)


## Exercice 2.11 — Groupby avancé et agrégations multiples
Objectif : Combiner plusieurs agrégations en une seule instruction. 

Instructions :

- Regroupez par Departement et Satisfait simultanément.
- Utilisez .agg() pour calculer, pour chaque groupe, la moyenne, l'écart-type et le nombre d'observations (count) du salaire.
- Renommez proprement les colonnes résultantes.<br> **Difficulté** : 🔴 Avancé

In [56]:
# Regroupement par Departement et Satisfait simultanement
# data_employee.groupby(
#     ["Departement", "Satisfait"]
#     )["Salaire_Mensuel_BIF"].agg(
#         ["mean", "std", "count"]
#     ).rename(columns={
#         "mean": "Salaire_Moyen_BIF",
#         "std": "Ecart_Type_BIF",
#         "count": "Nombre_Employes"
#     })

data_employee.groupby(
    ["Departement", "Satisfait"]
    ).agg(
        Salaire_Moyen_BIF=("Salaire_Mensuel_BIF", "mean"),
        Ecart_Type_BIF=("Salaire_Mensuel_BIF", "std"),
        Nombre_Employes=("Salaire_Mensuel_BIF", "count")
    )

Salaire_Moyen_BIF  Ecart_Type_BIF  \
Departement                Satisfait                                      
Administration             Non             2.419433e+06    2.009560e+06   
                           Oui             2.463582e+06    1.877438e+06   
Commercial                 Non             1.852059e+06    1.513937e+06   
                           Oui             2.441373e+06    1.850223e+06   
Finance                    Non             2.339170e+06    1.575326e+06   
                           Oui             3.887841e+06    2.122867e+06   
Informatique               Non             2.744276e+06    1.869428e+06   
                           Oui             2.876916e+06    1.892840e+06   
Logistique                 Non             2.373807e+06    1.611403e+06   
                           Oui             2.679350e+06    2.115461e+06   
Marketing                  Non             2.618188e+06    1.783509e+06   
                           Oui             2.394573e+06    1.813391e+06   
Production                 Non             2.053247e+06    1.451302e+06   
                           Oui             1.988304e+06    1.083490e+06   
Recherche et Développement Non             2.404858e+06    1.942758e+06   
                           Oui             2.831405e+06    1.632604e+06   
Ressources Humaines        Non             3.094174e+06    1.932558e+06   
                           Oui             3.004883e+06    1.719042e+06   
Support Client             Non             1.435510e+06    1.083195e+06   
                           Oui             2.352542e+06    1.585234e+06   

                                      Nombre_Employes  
Departement                Satisfait                   
Administration             Non                     18  
                           Oui                     32  
Commercial                 Non                     17  
                           Oui                     26  
Finance                    Non                     15  
                           Oui                     22  
Informatique               Non                     18  
                           Oui                     27  
Logistique                 Non                     24  
                           Oui                     21  
Marketing                  Non                     22  
                           Oui                     23  
Production                 Non                     20  
                           Oui                     34  
Recherche et Développement Non                     14  
                           Oui                     30  
Ressources Humaines        Non                     21  
                           Oui                     31  
Support Client             Non                     15  
                           Oui                     31

## Exercice 2.12 — Feature engineering (partie 1)
Objectif : Créer de nouvelles variables à partir des colonnes existantes. 

Instructions :

- Créez une colonne Salaire_Annuel_BIF à partir de Salaire_Mensuel_BIF.
- Créez une colonne catégorielle Tranche_Age (par exemple : "20-30", "31-40", "41-50", "51-60") à partir de Age, à l'aide de pd.cut.
- Sur rent_prediction.csv, créez une variable dérivée pertinente de votre choix (par exemple un prix au m²).<br> **Difficulté** : 🟡 Intermédiaire

In [61]:
# Création du salaire annuel
data_employee["Salaire_Annuel_BIF"] = data_employee["Salaire_Mensuel_BIF"] * 12
data_employee[["Salaire_Mensuel_BIF", "Salaire_Annuel_BIF"]].head()

,Salaire_Mensuel_BIF,Salaire_Annuel_BIF
0,2804177.0,33650124.0
1,6000000.0,72000000.0
2,4147951.0,49775412.0
3,896874.0,10762488.0
4,3062453.0,36749436.0


In [62]:
# Définition des intervalles
bornes = [20, 30, 40, 50, 60]

# Définition des étiquettes
labels = ["20-30", "31-40", "41-50", "51-60"]

# Création de la variable catégorielle
data_employee["Tranche_Age"] = pd.cut(
    data_employee["Age"],
    bins=bornes,
    labels=labels,
    include_lowest=True
)

data_employee[["Age", "Tranche_Age"]].head()

,Age,Tranche_Age
0,58.0,51-60
1,48.0,41-50
2,34.0,31-40
3,27.0,20-30
4,NaN,NaN


In [64]:
# Création du prix par mètre carré
data_rent["Prix_m2_BIF"] = data_rent["LoyerMensuel_BIF"] / data_rent["Superficie_m2"]
data_rent[["LoyerMensuel_BIF", "Superficie_m2", "Prix_m2_BIF"]].head()

,LoyerMensuel_BIF,Superficie_m2,Prix_m2_BIF
0,2600000,192.0,13541.666667
1,2600000,234.0,11111.111111
2,1069218,151.0,7080.913907
3,2600000,227.0,11453.744493
4,2085458,262.0,7959.763359


## Exercice 2.13 — Feature engineering (partie 2) — Encodage
Objectif : Préparer les variables catégorielles pour le Machine Learning (Partie 5). 

Instructions :

- Encodez la variable cible Satisfait en 0/1.
- Encodez les variables catégorielles ordinales (Niveau_Etudes, Equilibre_Vie_Travail, Satisfaction_Salaire) en respectant leur ordre logique (par exemple à l'aide d'un dictionnaire de correspondance appliqué avec .map()).
- Justifiez pourquoi ces colonnes ne doivent pas être encodées avec un simple one-hot encoding.<br> **Difficulté** : 🔴 Avancé

In [57]:
# Encodage de la variable cible Satisfait
data_employee["Satisfait"] = data_employee["Satisfait"].map({
    "Oui": 1,
    "Non": 0
})

### Encodage des variables ordinales

In [58]:
# Niveau_Etudes

niveau_etudes = {
    "Primaire": 1,
    "Secondaire": 2,
    "Licence": 3,
    "Master": 4,
    "Doctorat": 5
}

data_employee["Niveau_Etudes"] = data_employee["Niveau_Etudes"].map(niveau_etudes)

In [59]:
# Equilibre_Vie_Travail

equilibre = {
    "Très faible": 1,
    "Faible": 2,
    "Moyen": 3,
    "Bon": 4,
    "Excellent": 5
}

data_employee["Equilibre_Vie_Travail"] = data_employee["Equilibre_Vie_Travail"].map(equilibre)

In [60]:
# Satisfaction_Salaire

satisfaction_salaire = {
    "Très insatisfait": 1,
    "Insatisfait": 2,
    "Neutre": 3,
    "Satisfait": 4,
    "Très satisfait": 5
}

data_employee["Satisfaction_Salaire"] = data_employee["Satisfaction_Salaire"].map(satisfaction_salaire)

#### Justification

Les variables **Niveau_Etudes**, **Equilibre_Vie_Travail** et **Satisfaction_Salaire** sont des **variables ordinales**, c'est-à-dire que leurs catégories suivent un ordre logique.

Pour cette raison, un **encodage ordinal** est plus approprié qu'un **one-hot encoding**.

Le **one-hot encoding** transforme chaque catégorie en une colonne indépendante contenant des valeurs 0 ou 1. Cette méthode est adaptée aux variables **nominales**, mais elle supprime complètement l'information d'ordre existant entre les catégories.

En utilisant un encodage ordinal avec `.map()`, on conserve la hiérarchie naturelle des modalités, ce qui permet à de nombreux algorithmes de machine learning (régression, arbres de décision, gradient boosting, etc.) d'exploiter cette information pour améliorer leurs performances.

## Exercice 2.14 — Corrélation
Objectif : Identifier les relations linéaires entre variables numériques. 

Instructions :

- Calculez la matrice de corrélation des variables numériques (incluant votre version encodée de Satisfait) avec .corr().
- Identifiez les 3 variables les plus corrélées (positivement ou négativement) avec la satisfaction.
- Faites de même pour les variables numériques de rent_prediction.csv vis-à-vis du loyer.<br> **Difficulté** : 🟡 Intermédiaire

#### Pour employee_satisfaction.csv

In [66]:
# Calcul de la matrice de corrélation des variables numériques 

# Sélection des variables numériques
numeric_employee = data_employee.select_dtypes(include="number")

# Calcul de la matrice de corrélation
corr_employee = numeric_employee.corr()

corr_employee

,ID_Employe,Age,Annees_Experience,Niveau_Etudes,Salaire_Mensuel_BIF,Heures_Supplementaires,Heures_Formation,Anciennete_Entreprise,Nombre_Promotions,Evaluation_Performance,Equilibre_Vie_Travail,Satisfaction_Salaire,Nombre_Absences,Satisfait,Salaire_Annuel_BIF
ID_Employe,1.000000,0.070511,0.000792,0.018546,0.010398,0.010838,0.057981,-0.018214,0.006316,-0.052755,0.014188,0.061386,-0.002127,0.062501,0.010398
Age,0.070511,1.000000,0.565629,0.239626,0.536278,-0.057255,-0.066590,0.412267,0.232904,0.011705,0.023854,-0.031532,0.025984,0.026060,0.536278
Annees_Experience,0.000792,0.565629,1.000000,0.255108,0.917012,-0.028425,-0.019339,0.727453,0.458894,0.041506,0.018436,0.006337,0.037932,0.099438,0.917012
Niveau_Etudes,0.018546,0.239626,0.255108,1.000000,0.496044,-0.024284,-0.071536,0.192457,0.112386,-0.049664,-0.029901,0.067016,-0.034566,0.045198,0.496044
Salaire_Mensuel_BIF,0.010398,0.536278,0.917012,0.496044,1.000000,-0.015209,-0.050074,0.687872,0.408877,0.042140,0.007361,0.042916,0.020035,0.072207,1.000000
Heures_Supplementaires,0.010838,-0.057255,-0.028425,-0.024284,-0.015209,1.000000,0.056116,0.043742,0.023007,0.035876,-0.108344,0.025595,0.001780,-0.108439,-0.015209
Heures_Formation,0.057981,-0.066590,-0.019339,-0.071536,-0.050074,0.056116,1.000000,-0.038689,0.043173,0.038017,-0.026796,-0.004751,-0.012733,0.199115,-0.050074
Anciennete_Entreprise,-0.018214,0.412267,0.727453,0.192457,0.687872,0.043742,-0.038689,1.000000,0.640549,0.095008,0.056944,0.023279,0.013831,0.147564,0.687872
Nombre_Promotions,0.006316,0.232904,0.458894,0.112386,0.408877,0.023007,0.043173,0.640549,1.000000,0.092873,0.012848,0.065273,0.024489,0.219776,0.408877
Evaluation_Performance,-0.052755,0.011705,0.041506,-0.049664,0.042140,0.035876,0.038017,0.095008,0.092873,1.000000,-0.009638,0.023533,0.008786,0.153270,0.042140


In [68]:
# Identification des 3 variables les plus corrélées avec Satisfait

# Corrélations avec Satisfait
corr_satisfaction = corr_employee["Satisfait"].drop("Satisfait")

# Trier selon la force de la corrélation absolue
top_3_satisfaction = (
    corr_satisfaction
    .abs()
    .sort_values(ascending=False)
    .head(3)
)

top_3_satisfaction

Satisfaction_Salaire     0.468590
Equilibre_Vie_Travail    0.236345
Nombre_Promotions        0.219776
Name: Satisfait, dtype: float64

#### Pour rent_prediction.csv

In [71]:
# Sélection des variables numériques
numeric_rent = data_rent.select_dtypes(include="number")

# Matrice de corrélation
corr_rent = numeric_rent.corr()

corr_rent

,IdentifiantMaison,Chambres,Superficie_m2,DistanceRoute_m,AgeMaison,LoyerMensuel_BIF,Prix_m2_BIF
IdentifiantMaison,1.000000,0.033125,0.035433,0.049298,-0.066240,0.041912,-0.025532
Chambres,0.033125,1.000000,0.975543,0.051887,0.054955,0.623275,-0.312513
Superficie_m2,0.035433,0.975543,1.000000,0.072780,0.062029,0.615623,-0.356148
DistanceRoute_m,0.049298,0.051887,0.072780,1.000000,-0.070469,0.015409,-0.074722
AgeMaison,-0.066240,0.054955,0.062029,-0.070469,1.000000,0.022581,-0.107362
LoyerMensuel_BIF,0.041912,0.623275,0.615623,0.015409,0.022581,1.000000,0.331397
Prix_m2_BIF,-0.025532,-0.312513,-0.356148,-0.074722,-0.107362,0.331397,1.000000


In [73]:
# Corrélations avec le loyer
corr_rent_target = corr_rent["LoyerMensuel_BIF"].drop("LoyerMensuel_BIF")

# Top 3 variables les plus corrélées
top_3_rent = (
    corr_rent_target
    .abs()
    .sort_values(ascending=False)
    .head(3)
)

top_3_rent

Chambres         0.623275
Superficie_m2    0.615623
Prix_m2_BIF      0.331397
Name: LoyerMensuel_BIF, dtype: float64

## Exercice 2.15 — Export CSV
Objectif : Sauvegarder un dataset nettoyé et enrichi. 

Instructions :

- Exportez une version nettoyée (sans valeurs manquantes, sans doublons, avec vos nouvelles colonnes) de employee_satisfaction.csv.
- Nommez le fichier de façon explicite (par exemple employee_satisfaction_clean.csv).
- Vérifiez que le fichier exporté peut être rechargé correctement avec read_csv.<br> **Difficulté** : 🟢 Débutant

#### Pour rent_prediction.csv

In [ ]:
# Traitement des valeurs nulles pour les colonnes numeriques
colonnes_numeriques = [
    "Chambres",
    "Superficie_m2",
    "DistanceRoute_m",
    "AgeMaison"
]

for col in colonnes_numeriques:
    data_rent[col] = data_rent[col].fillna(data_rent[col].median())

In [76]:
# Traitement des valeurs nulles pour les colonnes numeriques
colonnes_categorielles = [
    "Salon",
    "SalleDeBainInterieure",
    "Parking",
    "Meuble",
    "Jardin",
    "Quartier"
]

for col in colonnes_categorielles:
    data_rent[col] = data_rent[col].fillna(data_rent[col].mode()[0])

In [78]:
# Recalcul de Prix_m2_BIF
data_rent["Prix_m2_BIF"] = (
    data_rent["LoyerMensuel_BIF"] /
    data_rent["Superficie_m2"]
)

# Verification des valeurs nulles apres nettoyage
data_rent.isnull().sum()

IdentifiantMaison        0
Chambres                 0
Salon                    0
SalleDeBainInterieure    0
Parking                  0
Meuble                   0
Jardin                   0
Superficie_m2            0
DistanceRoute_m          0
Quartier                 0
AgeMaison                0
LoyerMensuel_BIF         0
Prix_m2_BIF              0
dtype: int64

In [79]:
# Verification des doublons apres nettoyage
data_rent.duplicated().sum()

0

In [80]:
# Exportation du dataset nettoye
data_rent.to_csv(
    "../datasets/rent_prediction_clean.csv",
    index=False
)

In [83]:
# Test du chargement
test_rent = pd.read_csv("../datasets/rent_prediction_clean.csv")
test_rent.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 510 entries, 0 to 509
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   IdentifiantMaison      510 non-null    int64  
 1   Chambres               510 non-null    float64
 2   Salon                  510 non-null    object 
 3   SalleDeBainInterieure  510 non-null    object 
 4   Parking                510 non-null    object 
 5   Meuble                 510 non-null    object 
 6   Jardin                 510 non-null    object 
 7   Superficie_m2          510 non-null    float64
 8   DistanceRoute_m        510 non-null    float64
 9   Quartier               510 non-null    object 
 10  AgeMaison              510 non-null    float64
 11  LoyerMensuel_BIF       510 non-null    int64  
 12  Prix_m2_BIF            510 non-null    float64
dtypes: float64(5), int64(2), object(6)
memory usage: 51.9+ KB


#### Pour employee_satisfaction.csv

In [85]:
numeric_columns = data_employee.select_dtypes(
    include=["int64", "float64"]
).columns

for col in numeric_columns:
    data_employee[col] = data_employee[col].fillna(
        data_employee[col].median()
    )

In [86]:
categorical_columns = data_employee.select_dtypes(
    include=["object"]
).columns

for col in categorical_columns:
    data_employee[col] = data_employee[col].fillna(
        data_employee[col].mode()[0]
    )

In [87]:
data_employee["Tranche_Age"] = pd.cut(
    data_employee["Age"],
    bins=[20, 30, 40, 50, 60],
    labels=["20-30", "31-40", "41-50", "51-60"],
    include_lowest=True
)

In [88]:
data_employee["Equilibre_Vie_Travail"] = data_employee[
    "Equilibre_Vie_Travail"
].fillna(
    data_employee["Equilibre_Vie_Travail"].median()
)

In [90]:
data_employee = data_employee.drop_duplicates()
print(data_employee.duplicated().sum())

0


In [92]:
data_employee.to_csv(
    "../datasets/employee_satisfaction_clean.csv",
    index=False
)

In [93]:
test_employee = pd.read_csv("../datasets/employee_satisfaction_clean.csv")
test_employee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 510 entries, 0 to 509
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   ID_Employe              510 non-null    int64  
 1   Age                     510 non-null    float64
 2   Sexe                    510 non-null    object 
 3   Departement             510 non-null    object 
 4   Annees_Experience       510 non-null    float64
 5   Niveau_Etudes           510 non-null    float64
 6   Salaire_Mensuel_BIF     510 non-null    float64
 7   Travail_A_Distance      510 non-null    object 
 8   Heures_Supplementaires  510 non-null    float64
 9   Heures_Formation        510 non-null    float64
 10  Anciennete_Entreprise   510 non-null    float64
 11  Nombre_Promotions       510 non-null    float64
 12  Evaluation_Performance  510 non-null    float64
 13  Equilibre_Vie_Travail   510 non-null    float64
 14  Satisfaction_Salaire    510 non-null    fl